### PR TITLE
Add Prometheus Exporter

### PR DESCRIPTION
**Pull request by esigo**
This PR attempts to finish #263
## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [x] `CHANGELOG.md` updated for non-trivial changes
* [x] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

**Review by lalitb — APPROVED** [review]
Looks good in general.

File: exporters/prometheus/include/opentelemetry/exporters/prometheus/prometheus_exporter.h [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+#  include "prometheus/exposer.h"
+
+/**
+ * This class is an implementation of the MetricsExporter interface and
+ * exports Prometheus metrics data. Functions in this class should be
+ * called by the Controller in our data pipeline.
+ */
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdkmetrics = opentelemetry::sdk::metrics;
```
[the remark this answers is not in the document]
Comment (esigo) [reply exporters/prometheus/include/opentelemetry/exporters/prometheus/prometheus_exporter.h]: done

File: CHANGELOG.md [hunk 2]
```diff
@@ -19,6 +19,7 @@ Increment the:
 
 ## [1.0.1] 2021-10-21
 
+* [EXPORTER] Prometheus Exporter ([#1031](https://github.com/open-telemetry/opentelemetry-cpp/pull/1031))
```
Comment (ThomsonTan) [inline CHANGELOG.md:22]: This should be put into the above section for "Unreleased".

File: ci/do_ci.sh [hunk 3]
```diff
@@ -85,6 +84,7 @@ elif [[ "$1" == "cmake.c++20.stl.test" ]]; then
   make test
   exit 0
 elif [[ "$1" == "cmake.legacy.test" ]]; then
+
```
Comment (ThomsonTan) [inline ci/do_ci.sh:87]: nit: remove this empty line.

File: ci/do_ci.sh [hunk 4]
```diff
@@ -93,6 +93,7 @@ elif [[ "$1" == "cmake.legacy.test" ]]; then
   cmake -DCMAKE_BUILD_TYPE=Debug  \
         -DCMAKE_CXX_FLAGS="-Werror" \
         -DCMAKE_CXX_STANDARD=11 \
+        -DCMAKE_EXE_LINKER_FLAGS="-lpthread" \
```
Comment (ThomsonTan) [inline ci/do_ci.sh:96]: Why is this required?

**Comment by lalitb** [comment]
@esigo - Sorry about this, but realized afterward that the #1053 merge will break this PR. Unfortunately, we need changes in this PR for CI to go through ( I can do it if you want in your branch as my PR was the culprit to break this ).

File: exporters/prometheus/CMakeLists.txt [hunk 5]
```diff
@@ -15,16 +15,13 @@
 include_directories(include)
 find_package(prometheus-cpp CONFIG REQUIRED)
 
-add_library(prometheus_exporter_deprecated src/prometheus_collector.cc
```
Comment (ThomsonTan) [inline exporters/prometheus/CMakeLists.txt:18]: The target is removed here, but I didn't see it is added back in other places but it is referenced below in `target_include_directories`.

**Comment by esigo** [comment]
> @esigo - Sorry about this, but realized afterward that the #1053 merge will break this PR. Unfortunately, we need changes in this PR for CI to go through ( I can do it if you want in your branch as my PR was the culprit to break this ).
@lalitb No problem. I'll take care of it :)

File: exporters/prometheus/include/opentelemetry/exporters/prometheus/prometheus_exporter.h [hunk 6]
```diff
@@ -0,0 +1,111 @@
+/*
```
Comment (ThomsonTan) [inline exporters/prometheus/include/opentelemetry/exporters/prometheus/prometheus_exporter.h:1]: Simplify the copyright header as below?

```cpp
// Copyright The OpenTelemetry Authors
// SPDX-License-Identifier: Apache-2.0
```

File: CHANGELOG.md (hunk 2 above)
Comment (esigo) [reply CHANGELOG.md]: done

File: ci/do_ci.sh (hunk 3 above)
Comment (esigo) [reply ci/do_ci.sh]: done

File: ci/do_ci.sh (hunk 4 above)
Comment (esigo) [reply ci/do_ci.sh]: was not needed. cleaned

File: exporters/prometheus/CMakeLists.txt (hunk 5 above)
Comment (esigo) [reply exporters/prometheus/CMakeLists.txt:18]: fixed

File: exporters/prometheus/include/opentelemetry/exporters/prometheus/prometheus_exporter.h (hunk 6 above)
Comment (esigo) [reply exporters/prometheus/include/opentelemetry/exporters/prometheus/prometheus_exporter.h]: done
